File: leetcode2529/leetcode2529.cpp
#include<bits/stdc++.h>
using namespace std;

class Solution {
    int lower_bound(vector<int>& nums, int target){
        int left = 0, right = nums.size() - 1;
        while (left <= right)
        {
            int mid = left + (right - left) / 2;
            if (nums[mid] >= target){
                right = mid - 1;
            }else{
                left = mid + 1;
            }
        }
        return left;
    }
    public:
        int maximumCount(vector<int>& nums) {
            int index1 = lower_bound(nums, 1);
            int pos = nums.size() - index1;
            int index2 = lower_bound(nums, 0);
            int neg = index2;
            return max(pos, neg);
        }
    };

File: leetcode2529/leetcode2529.py
class Solution:
    def low_bound(self, nums, target):
        left, right = 0, len(nums) - 1
        while left <= right:
            mid = (left + right) // 2
            if (nums[mid] >= target):
                right = mid - 1
            else:
                left = mid + 1
        return left
    def maximumCount(self, nums: List[int]) -> int:
        pos = len(nums) - self.low_bound(nums, 1)
        neg = self.low_bound(nums, 0)
    
        return max(pos, neg)
